packages/next/taskfile-webpack.js

```diff
@@ -26,13 +26,20 @@ module.exports = function (task) {
           })
         }
 
-        if (process.env.ANALYZE_STATS) {
+        if (process.env.ANALYZE_WEBPACK_STATS) {
           require('fs').writeFileSync(
             require('path').join(__dirname, options.name + '-stats.json'),
             JSON.stringify(stats.toJson())
           )
         }
 
+        if (process.env.PRINT_WEBPACK_WARNINGS) {
```

Comment:
Why not always include them? Otherwise they'll never surface automatically.

Reply:
Not wait, this does point to a potential mistake. Shouldn't the dev bundles be built in development mode?
```diff
-    mode: 'production',
+    mode: dev ? 'development' : 'production',
```
Was a hard coded `'production'` selected intentionally for some reason?

Reply:
I don't know. Does git-blame help identify when this was introduced?

Reply:
Yes, the split between prod and dev bundles was first introduced in #52997, then reverted in #55117, and re-landed in #55139. The PR description in #52997 sounds to me like this was done intentionally to also optimise the performance of `next dev`. @feedthejim Is that assumption correct?

Reply:
Hm, interesting, despite the explizit mode-independent definition of the `optimization` config, and despite seemingly all relevant options that are otherwise [set via the production mode](https://webpack.js.org/configuration/mode/#usage) being included, there is indeed a bundle size difference:
||production|development|
|-|-|-|
|app-page.runtime.dev.js|633 KiB|662 KiB|
|app-page-experimental.runtime.dev.js|666 KiB|696 KiB|
|pages.runtime.dev.js|63.9 KiB|81.5 KiB|